### PR TITLE
Removed bug producing insert from JSessionStorageDatabase::write()

### PR DESCRIPTION
**Pull request by ghost**
If update results in an error[such as a non existant row] then write will already return false.  If however it succeeds but is identical to the current data, affectedRows will be 0, which then invokes insert while will ALWAYS fail because it is a duplicate row.

An insert is already performed when the session is created in JSession, there is no need for this one and it can only result in an error.

### CODE REVIEW

File: libraries/joomla/session/storage/database.php [hunk 1]
```diff
@@ -117,13 +117,16 @@ class JSessionStorageDatabase extends JSessionStorage
...
-			// If the session does not exist, we need to insert the session.
-			$db->setQuery(
-				'INSERT INTO ' . $db->quoteName('#__session') .
-				' (' . $db->quoteName('session_id') . ', ' . $db->quoteName('data') . ', ' . $db->quoteName('time') . ')' .
-				' VALUES (' . $db->quote($id) . ', ' . $db->quote($data) . ', ' . (int) time() . ')'
-			);
-			return (boolean) $db->query();
+            // update did not error, so it succeeded but was identical
+            if (JDEBUG) {
+                jimport('joomla.error.log');
```
Comment (LouisLandry) [inline libraries/joomla/session/storage/database.php:122]: This is deprecated usage of JLog.  You'll want to have a look at `JPATH_PLATFORM/joomla/log/log.php`.  The syntax is something like:

``` php
<?php
JLog::add(sprintf('Session ID %s update is identical to current record', $id));
```

**Comment by elkuku** [comment]
**Checkstyle error details:**
libraries/joomla/session/storage/database.php:120, 121
Tabs must be used to indent lines; spaces are not allowed

libraries/joomla/session/storage/database.php:121
Expected "if (...)\n"; found "if (...) "

libraries/joomla/session/storage/database.php:122 - 129
Tabs must be used to indent lines; spaces are not allowed

**Comment by eddieajau** [comment]
Any followup on this issue?

**Comment by chdemko** [comment]
@garyamort The pull request cannot be merged

**Comment by AmyStephen** [comment]
@garyamort - It would be good to get this fix in-it continues to be reported as an error, especially on high-volume sites.

http://joomlacode.org/gf/project/joomla/tracker/?action=TrackerItemEdit&tracker_item_id=28510

http://joomlacode.org/gf/project/joomla/tracker/?action=TrackerItemEdit&tracker_item_id=28249

**Comment by ghost** [comment]
hmm, thought this was fixed in another pull request...  Elin reminded me it's been sitting here.  Ok, I';ll have to pull the latest code and redo the update, this time without using depreciated JLog.

**Comment by ghost** [comment]
I re-arranged my repos into orgs for my own clarity and this branch was lost.   I've recreated the fix and submitted it as pull request 1209.
https://github.com/joomla/joomla-platform/pull/1209

**Pull request closed without merging**